billing/views.py

```diff
@@ -54,38 +53,40 @@ def invoice_payment_failed(self, invoice: stripe.Invoice) -> None:
                 stripe_subscription_id=invoice.subscription,
             ),
         )
-        updated = Owner.objects.filter(
+        updated: int = Owner.objects.filter(
             stripe_customer_id=invoice.customer,
             stripe_subscription_id=invoice.subscription,
         ).update(delinquent=True)
         self._log_updated(updated)
 
     def customer_subscription_deleted(self, subscription: stripe.Subscription) -> None:
-        try:
+        log.info(
+            "Customer Subscription Deleted - Setting free plan and deactivating repos for stripe customer",
+            extra=dict(
+                stripe_subscription_id=subscription.id,
+                stripe_customer_id=subscription.customer,
+            ),
+        )
+        owners: QuerySet[Owner] = Owner.objects.filter(
+            stripe_customer_id=subscription.customer,
+            stripe_subscription_id=subscription.id,
+        )
+        if not owners.exists():
             log.info(
-                "Customer Subscription Deleted - Setting free plan and deactivating repos for stripe customer",
+                "Customer Subscription Deleted - Couldn't find owner, subscription likely already deleted",
                 extra=dict(
                     stripe_subscription_id=subscription.id,
                     stripe_customer_id=subscription.customer,
                 ),
             )
-            owner: Owner = Owner.objects.get(
-                stripe_customer_id=subscription.customer,
-                stripe_subscription_id=subscription.id,
-            )
-            plan_service = PlanService(current_org=owner)
+            return
+
+        for owner in owners:
+            plan_service = PlanService(owner)
```

Comment:
Could you fill me in on the removal of current_org = owner in lieu of owner?

Reply:
Didn't even notice this lol. The `PlanService` constructor only takes one argument (`current_org`), so either way works. Happy to change to the other way if you'd prefer!

Reply:
for consistency's sake, and for helping our future self out, I think we should update 😄 

(There's always a possibility we add another optional param to this function too)

Reply:
Updated @ajay-sentry!